x-pack/plugins/file_data_visualizer/public/plugin.ts

```diff
@@ -40,7 +42,9 @@ export class FileDataVisualizerPlugin
       FileDataVisualizerSetupDependencies,
       FileDataVisualizerStartDependencies
     > {
-  public setup() {}
+  public setup(core: CoreSetup, plugins: FileDataVisualizerSetupDependencies) {
+    setSetupServices(plugins);
```

Comment:
@elastic/kibana-security Would it be possible to expose authc on start as well as setup? File data visualizer uses authc outside of plugin setup.

@jgowdyelastic From my understanding, setup services should only be used during setup and not after the plugin has started.

Reply:
@nreese No objections from me. IMO the services exposed by `authc` are more appropriate for the `start` phase anyway

Reply:
`authc` is just used here to get the current username.
@elastic/kibana-security what is the correct way to get the current user?

Reply:
`authc.getCurrentUser()` is the best way to get the current user

Reply:
Probably. Follow along here: https://github.com/elastic/kibana/pull/98307

Reply:
https://github.com/elastic/kibana/pull/98307 has merged. Backports are in progress